### PR TITLE
fix: Allow `<reference />` to self in UMD tests

### PR DESCRIPTION
**Pull request by ExE-Boss**
It&nbsp;has&nbsp;been decided&nbsp;in&nbsp;#708&nbsp;that tests&nbsp;of&nbsp;**UMD**&nbsp;globals will&nbsp;require:
```ts
/// <reference types="<package_name>"/>
```
at&nbsp;the&nbsp;top.

This&nbsp;makes the&nbsp;above&nbsp;allowed.

---

Note&nbsp;that:
```ts
/// <reference types="<package_name>"/>
import foo = require("<package_name>");
```
is&nbsp;still&nbsp;correctly&nbsp;disallowed.

### CODE REVIEW

File: src/lib/module-info.ts [hunk 1]
```diff
@@ -355,18 +358,30 @@ export function getTestDependencies(
...
+        if (hasImports && referencesSelf) {
+            throw new Error(`'${filePath()}' unnecessarily references the package. This can be removed.`);
+        }
+
+        // TODO: test for exports as well and uncomment this in a future release
+        /*
+        if (!hasImports && !hasExports && !referencesSelf) {
+            throw new Error(`UMD test '${filePath()}' must specify '/// <reference types="${packageName}"/>'`);
+        }
+        */
```
Comment (ExE-Boss) [inline src/lib/module-info.ts]: I’ve&nbsp;also prepared&nbsp;code that&nbsp;will&nbsp;make `/// <reference types="${packageName}"/>` mandatory&nbsp;for&nbsp;tests of&nbsp;**UMD**&nbsp;globals.

Comment (sandersn) [reply src/lib/module-info.ts]: I think I'm OK with testing *either* the namespace or the import for UMDs.

Comment (ExE-Boss) [reply src/lib/module-info.ts]: This&nbsp;won’t&nbsp;add a&nbsp;requirement to&nbsp;have a&nbsp;global&nbsp;test of&nbsp;**UMD**&nbsp;modules.

File: src/lib/module-info.ts [hunk 2]
```diff
@@ -355,18 +359,37 @@ export function getTestDependencies(
...
             if (!imported.startsWith(".")) {
                 const dep = rootName(imported, typeFiles);
                 if (!dependencies.has(dep) && dep !== packageName) {
                     testDependencies.add(dep);
                 }
             }
         }
+
+        isModule = hasImports || (() => {
+            // FIXME: This results in files without imports to be walked twice,
```
Comment (sandersn) [inline src/lib/module-info.ts]: good to note, although performance hasn't been a problem yet, and I just got back 35% or so from removing `async`.

**Comment by sandersn** [comment]
It's now in the production branch, so Definitely Typed's CI will get the new whitelist next time it runs.

**Pull request merged**